napari/layers/image/experimental/octree_intersection.py

```diff
@@ -0,0 +1,73 @@
+"""OctreeIntersection class.
+"""
+from typing import Tuple
+
+import numpy as np
+
+from .octree_util import OctreeLevelInfo
+
+# TODO_OCTREE: These types might be a horrible idea but trying it for now.
+Float2 = np.ndarray  # [x, y] dtype=float64 (default type)
```

Comment:
yeah, personally I don't love this and we don't do it in many other places, but not need to change right now

Reply:
Yeah I'm not sure what our convention is for `ndarray`. If the code expects certain dimensions or a certain `dtype` where should that be expressed in comments and/or types? Is there a guide maybe from other scipy projects if not napari. Seems like places we use a lot of `ndarray` are also places with very little typing. Not sure if that's on purpose or not?

Reply:
> Whatever is napari and/or scipy convention I want to do that.

I just did some quick poking around. Nothing comprehensive, but gives some rough ideas

In numpy you might see methods with names like `_assert_2d` at the beginning of functions see here https://github.com/numpy/numpy/blob/master/numpy/linalg/linalg.py#L539 and it's definition here https://github.com/numpy/numpy/blob/43683b3256a86659f230dcadbcde1f8020398bfa/numpy/linalg/linalg.py#L193 that will do that sort of dimensionality checking

In PyTorch you can see here an example of computing the length of something and checking it here https://github.com/pytorch/pytorch/blob/master/torch/nn/grad.py#L19-L21

We've also done that to a limited extent within napari already - see here for example, https://github.com/napari/napari/blob/master/napari/layers/shapes/_shapes_models/line.py#L50-L55.

So it seems like the conventions are more along the lines of "receiver assert the ndarray is the expected shape or type"